server.js

```diff
@@ -17,93 +17,124 @@ client.on('error', err => console.error(err));
 
 // ////// call the handler function 
 
-server.get('/location',locationHandler);
+server.get('/location', locationHandler);
 server.get('/weather', weatherHanddler);
 server.get('/events', eventHanddler); ////// here i have a problem and gorob help me to solve it //////
 
 // /////// location handler //////
-  function locationHandler(req,res){
-getlocation(req.query.data)
-.then(locationData => res.status(200).json(locationData));
-  };
+function locationHandler(req, res) {
+  const city = req.query.data;
+  getlocation(city)
+    // .then(locationData => res.status(200).json(locationData));
+    .then(data => sendJson(data, res))
+    .catch((error) => errorHandler(error, req, res));
+};
+
 // ///// weather handler ////////
-  function weatherHanddler(req,res) {
-    getWeather(req.query.data)
-       .then (weatherData => res.status(200).json(weatherData) );
-   };
+function weatherHanddler(req, res) {
+
+  getWeather(req.query.data)
+    .then(weatherData => res.status(200).json(weatherData));
+};
+
+
 // ///// event handler /////////////// 
-   function eventHanddler(req,res) {
-    getEventINFO(req.query.data)
-       .then (eventData => res.status(200).json(eventData) );
-   };
+function eventHanddler(req, res) {
+  getEventINFO(req.query.data)
+    .then(eventData => res.status(200).json(eventData));
+};
 // ///// get the data from API for location /////
-  function getlocation (city){
-      const url =`https://maps.googleapis.com/maps/api/geocode/json?address=${city}&key=${process.env.GEOCODE_API_KEY}`
-      return superagent.get(url)
-      .then( data =>{
-          return new Location (city , data.body)
-      })
-  };
-  // /////  get the data from API for weather /////
-  function getWeather (query) {
-    const url = `https://api.darksky.net/forecast/${process.env.DARKSKY_API_KEY}/${query.latitude},${query.longitude}`;
-    return superagent.get(url)
-    .then( data => {
+function getlocation(city) {
+  let SQL = 'select * FROM location WHERE search_query = $1 ';
```

Comment:
In my database, the table name is `locations` not `location` you should check that on your machine too.